examples/simple.rs

```diff
@@ -0,0 +1,223 @@
+
+    // Iterates over the glyphs in the GlyphRun
+    for glyph in glyph_run.glyphs() {
+        let glyph_id: GlyphId = glyph.id;
+        let glyph_x = run_x + glyph.x;
+        let glyph_y = run_y - glyph.y;
+        run_x += glyph.advance;
+        let Some(rendered_glyph) = render_glyph(
+            context,
+            &font_ref,
+            font_size,
+            glyph_id,
+            normalized_coords,
+            glyph_x.fract(),
+            glyph_y.fract(),
+        ) else {
+            println!("No glyph");
+            continue;
+        };
+
+        let glyph_width = usize::try_from(rendered_glyph.placement.width).expect("usize < 32 bits");
+        let glyph_height =
+            usize::try_from(rendered_glyph.placement.height).expect("usize < 32 bits");
+        let glyph_origin_x =
+            glyph_x.floor() as i32 + rendered_glyph.placement.left + padding as i32;
+        let glyph_origin_y =
+            (glyph_y.floor() as i32) - rendered_glyph.placement.top + padding as i32;
+
+        match rendered_glyph.content {
+            Content::Mask => {
+                let mut i = 0;
+                for off_y in 0..glyph_height as i32 {
+                    for off_x in 0..glyph_width as i32 {
+                        let x = (glyph_origin_x + off_x) as u32;
+                        let y = (glyph_origin_y + off_y) as u32;
+                        let alpha = rendered_glyph.data[i];
+                        if alpha > 0 {
+                            let color = Rgba::from([color.r, color.g, color.b, alpha]);
+                            img.get_pixel_mut(x, y).blend(&color);
+                        }
+                        i += 1;
+                    }
+                }
+            }
+            Content::SubpixelMask => unimplemented!(),
+            Content::Color => {
+                for (off_y, row) in rendered_glyph
+                    .data
+                    .chunks_exact(glyph_width * 4)
+                    .enumerate()
+                {
+                    for (off_x, pixel) in row.chunks_exact(4).enumerate() {
+                        let &[r, g, b, a] = pixel else {
+                            panic!("Not RGBA")
+                        };
+                        let x = (glyph_origin_x + off_x as i32) as u32;
+                        let y = (glyph_origin_y + off_y as i32) as u32;
+                        if a > 0 {
+                            let color = Rgba::from([r, g, b, a]);
+                            img.get_pixel_mut(x, y).blend(&color);
+                        }
+                    }
+                }
+            }
+        };
+    }
+}
+
+/// Render a glyph using Swash
+fn render_glyph(
+    context: &mut ScaleContext,
+    font: &FontRef,
+    font_size: f32,
+    glyph_id: GlyphId,
+    normalized_coords: &[NormalizedCoord],
+    x: f32,
+    y: f32,
+) -> Option<SwashImage> {
+    use zeno::{Format, Vector};
+
+    // Build the scaler
+    let mut scaler = context
```

Comment:
Nit: the scaler settings are dependent on glyph run properties but invariant per glyph so we could build it once per run and pass it in here. This is what I would suggest in real code as a performance optimization so the example might want to reflect that.

Reply:
Fixed